### DISCUSSION TITLE
How to do best slow motion effekt

### DISCUSSION THREAD

**Discussion by ia97lies** [dyn4j/dyn4j#301] [Q&A]
I wondered how to implement a slow motion properly with dyn4j? Can I just make the step time smaller? Ideas?

**Comment by wnbittle** [comment]
Sorry for the delay in responding.  You can slow the simulation by using:

```java
// instead of this:
// boolean stepped = this.world.update(elapsedTime);

// do this:
double invHz = this.world.getSettings().getStepFrequency();
double slowMotionFactor = 0.1;
boolean stepped = this.world.update(elapsedTime, invHz * slowMotionFactor, 1);
```

The simulation will still run at the same rate (defined by the step frequency which is 60 times per second by default), but the second parameter tells the engine that only `elapsedTime * slowMotionFactor` time has elapsed.

Another way, depending on your scenario would be to track `elapsedTime` yourself and handle it directly by using the `step` methods.  You can use the `update` methods as an example of how to track time.

**Reply by ia97lies** [reply]
I'm a little bit late with my answer too :D Thank you very much!
